Fix collision extraction from actor's data

AngheloAlf opened the pull request:
Fixes a crash when ZAPD tries to extract a CollisionHeader from overlay's data. 

The issue was caused because ZAPD didn't account the base address of the overlay.

The issue can be replicated creating an xml for the overlay `ovl_En_Jsjutan` and the following:
- `BaseAddress="0x80A897A0" RangeStart="0x12F8" RangeEnd="0x4C88" Segment="128"`
- `<Collision Name="D_80A8E400" Offset="0x4C60"/>`

AngheloAlf committed:
Fix collision extraction if it is in actor's data

Signed-off-by: Angie <angheloalf95@gmail.com>

mzxrules commented:
Most of the changes seem good. The main issue I have is that this pattern...

```c
	segmentNumber = GETSEGNUM(polySegmentOffset);
	if (segmentNumber == 0x80) { // Is defined in code?
		polyAddr -= SEG2FILESPACE(parent->baseAddress);
	}
```
...should ideally be part of SE2FILESPACE. The laziest/quickest thing to do would probably to just make it into it's own separate function

AngheloAlf commented:
I just followed the same pattern as in `ZDisplayList` (Example [1](https://github.com/zeldaret/ZAPD/blob/d9ba72302b1105d79a06c7da8d9c1bbb8ba2fceb/ZAPD/ZDisplayList.cpp#L801), [2](https://github.com/zeldaret/ZAPD/blob/d9ba72302b1105d79a06c7da8d9c1bbb8ba2fceb/ZAPD/ZDisplayList.cpp#L823) and [3](https://github.com/zeldaret/ZAPD/blob/d9ba72302b1105d79a06c7da8d9c1bbb8ba2fceb/ZAPD/ZDisplayList.cpp#L905)).

I'll try and add this check into `SEG2FILESPACE` macro.

AngheloAlf committed:
Seg2Filespace to avoid duplicated code
and rename the macro SEG2FILESPACE to GETSEGOFFSET

Also, try to use the macros GETSEGOFFSET and GETSEGNUM as much as
possible.

Signed-off-by: angie <angheloalf95@gmail.com>

AngheloAlf committed:
Reverse variables names

Signed-off-by: angie <angheloalf95@gmail.com>

AngheloAlf commented:
It should be fixed now.
